connector.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2020 Snowflake Computing Inc. All right reserved.
+
+package gosnowflake
+
+import (
+	"context"
+	"database/sql/driver"
+)
+
+// SnowflakeDriverInterface is the interface for a Snowflake driver
```

Comment:
I think the naming can be made less redundant since there shouldn't be conflicts within the same package.

Here's what I'm thinking:

```go
type Driver interface { // currently SnowflakeDriverInterface
  Open(...)
  OpenWithConfig(...)
}

type Connector struct { // currently SnowflakeConnector
  driver Driver
}

// We can't actually change this due to backwards compatibility. "SnowflakeDriver" is
// redundant and now we're stuck with the name unless we bump a major driver version.
type RestfulDriver struct {} // currently SnowflakeDriver
```

Reply:
Yep, if everyone is onboard with renaming the existing struct.
This would probably require a major version bump to not break back-compat

Reply:
Alternatively we could keep `SnowflakeDriver` as is. `SnowflakeConnector` and `SnowflakeDriverInterface` are both new in this PR so no compat issue with them.

Reply:
Kept the existing names as is, moved all new interface definition to connector.go
Renamed types to be shorter to be more idiomatic Golang